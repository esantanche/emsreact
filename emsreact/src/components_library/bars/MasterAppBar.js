import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import AppBar from '@material-ui/core/AppBar';

// Material UI uses withStyles to inject an array of styles into the DOM as CSS, using JSS
import { withStyles } from "@material-ui/core/styles/index";

// This array of style gets injected into the component as "classes"
const styles = {
    MasterAppBar: {
        borderRadius: "4px",
    }
};

/**
 * The AppBar in the header. Its children are the home button and the menu button.
 *
 * @param {object} classes Classes injected by withStyles function
 */
class MasterAppBar extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired
    };

    render() {

        const { classes } = this.props;

        return (

            <AppBar position="sticky" className={classes.MasterAppBar}>

                {this.props.children}

            </AppBar>

        )
    }

}

export default withStyles(styles)(MasterAppBar);

