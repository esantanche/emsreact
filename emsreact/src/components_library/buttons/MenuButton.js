import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Button from "@material-ui/core/Button";

// Material UI icons
import MenuIcon from "@material-ui/icons/Menu";

// Material UI functions that make classes available
import { withStyles } from "@material-ui/core/styles/index";

const styles = {
    button: {
        float: "right"
    }
};

/**
 * This is the menu button, the one made of three horizontal lines
 *
 * Props
 * @param {function} onclick Function to call when the user clicks on the menu button
 */
class MenuButton extends Component {

    static propTypes = {
        onclick: PropTypes.func
    };

    render() {

        const { onclick, classes } = this.props;

        return (

            <Button onClick={onclick}
                    className={classes.button}>
                <MenuIcon />
            </Button>

        );
    }
}

export default withStyles(styles)(MenuButton);