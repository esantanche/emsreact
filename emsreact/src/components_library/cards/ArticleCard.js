// Just a test to refactor
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI functions that make classes and width available
import { withStyles } from "@material-ui/core/styles/index";
import withWidth from '@material-ui/core/withWidth';

// Material UI components
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import CardActions from '@material-ui/core/CardActions';
import IconButton from '@material-ui/core/IconButton';

// Material UI icons
import MoreHoriz from '@material-ui/icons/MoreHoriz';

// App configuration
import { APP_CONFIGURATION } from '../../appConfiguration';

// Library used to strip html tags from text
import striptags from 'striptags';

// Used to cut text and add '...'
import Dotdotdot from 'react-dotdotdot';

// From component library
import SeparatorPane from '../panes/SeparatorPane';

// Helper function that takes a title and produces a slug
// which is how we put the title in the url
import { title_to_slug } from '../../helpers/title_to_slug';

const styles = {
    card_header: {
        height: "6em",
        display: "flex",
        alignItems: "center",
        paddingLeft: "24px",
        paddingRight: "24px"
    },
    container: {
        display: "flex",
        flexDirection: "row",
        alignItems: "stretch"
    },
    image_responsive_size: {
        height: "0",
        paddingTop: '100%'
    }
};

class ArticleCard extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired,
        title: PropTypes.string,
        image: PropTypes.string,
        text_content: PropTypes.string,
        article_node_id: PropTypes.number
    };

    produce_jsx_fragment_for_image_and_text_according_to_width(width, classes, image, text_content) {

        const image_fixed_size_according_to_width = {
            sm: "150px",
            md: "150px",
            lg: "180px",
            xl: "200px"
        };

        // We are going to remove any html tag from the text of the article
        // because we want no html tags when showing the text in the article
        // card
        const stripped_text_content = striptags(text_content);

        // When fetching images' url from the Drupal backend, we get relative paths
        // and we have to prepend the backend root url to access the images
        // The module Pathologic in Drupal makes urls absolutes, but not in this case

        if (width === 'xs') {

            return (
                <React.Fragment>

                    <Grid item xs={12}>

                        <CardMedia
                            className={classes.image_responsive_size}
                            image={APP_CONFIGURATION.backendUrl + image}
                            title=""
                        />

                    </Grid>

                    <SeparatorPane/>

                    <Grid item xs={12}>

                        <Typography variant="body2" component="div">

                            <Dotdotdot clamp={5}>

                                {stripped_text_content}

                            </Dotdotdot>

                        </Typography>

                    </Grid>

                </React.Fragment>
            );

        } else {

            // In this case we want the text to be as high as the image.
            // The dotdotdot component will cut the text and add '...'
            // at the end when the text is as high as the image

            return (
                <React.Fragment>

                    <CardMedia
                        style={{ height: image_fixed_size_according_to_width[width],
                                 width: image_fixed_size_according_to_width[width] }}
                        // className={classes.image_fixed_size}
                        image={APP_CONFIGURATION.backendUrl + image}
                        title=""
                    />

                    <Typography variant="body2" component="div"
                                style={{ flex: "1", height: image_fixed_size_according_to_width[width], marginLeft: "10px" }}>

                        <Dotdotdot clamp={image_fixed_size_according_to_width[width]}
                                   style={{ height: image_fixed_size_according_to_width[width] }}>

                            {stripped_text_content}

                        </Dotdotdot>

                    </Typography>

                </React.Fragment>
            );

        }

    }

    render() {

        const title_variant_according_to_width = {
            xs: "title",
            sm: "title",
            md: "subheading",
            lg: "title",
            xl: "title"
        };

        const { classes, title, width, image, text_content, article_node_id } = this.props;

        const jsx_fragment_for_image_and_text_according_to_width =
            this.produce_jsx_fragment_for_image_and_text_according_to_width(width, classes, image, text_content);

        // FIXME link to article to fix

        const slug_for_link_to_article = title_to_slug(title);

        return (

            <Card>

                {/*Used a div because the component CardHeader uses a headline variant of Typography while I    */}
                {/*need the possibility to change font size*/}
                {/*style={{ height: "6em", display: "flex", alignItems: "center", paddingLeft: "24px", paddingRight: "24px" }}*/}

                <div className={classes.card_header}>

                    <Typography variant={title_variant_according_to_width[width]}>
                        {title}
                    </Typography>

                </div>

                <CardContent>

                    <Grid container spacing={0} className={classes.container}>

                        {jsx_fragment_for_image_and_text_according_to_width}

                    </Grid>

                </CardContent>

                <CardActions>
                    <IconButton style={{ marginLeft: "auto" }} href={"/article/" + article_node_id + "/" + slug_for_link_to_article}>
                        <MoreHoriz/>
                    </IconButton>
                </CardActions>

            </Card>

        )

    }

}

export default withWidth()(withStyles(styles)(ArticleCard));
