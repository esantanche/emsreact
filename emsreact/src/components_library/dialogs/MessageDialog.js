import React, { Component } from 'react';
import PropTypes from 'prop-types'

// Material UI components
import Button from "@material-ui/core/es/Button/Button";
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

// Material UI icons
import ErrorOutline from '@material-ui/icons/ErrorOutline';

// On small devices, we want the dialog to go full screen
// This function does the trick
import withMobileDialog from '@material-ui/core/withMobileDialog';

// Material UI function that make classes available
import {withStyles} from "@material-ui/core/styles";

const styles = {
    dialog: {
        display: "flex",
        flexDirection: "column",
        alignItems: "center"
    },
    dialog_title: {
        textAlign: "center"
    },
    error_outline: {
        fontSize: "200%"
    }
};

/**
 * Message dialog to notify successful completion of operations.
 * Used as well for errors. *
 *
 * Props
 * @param {boolean} open True is the dialog should be visible
 * @param {boolean} it_is_an_error_message True if the dialog contains an error message
 * @param {string} title Dialog title
 * @param {string} message Message to display
 * @param {function} onclick Function to call when the user click on OK
 * @param {boolean} fullScreen True if the dialog has to go full screen. This prop
 * is provided by the function withMobileDialog. It will make the dialog full screen
 * on small devices.
 */
class MessageDialog extends Component {

    static propTypes = {
        open: PropTypes.bool.isRequired,
        it_is_an_error_message: PropTypes.bool,
        title: PropTypes.string,
        message: PropTypes.string,
        onclick: PropTypes.func,
        fullScreen: PropTypes.bool.isRequired
    };

    render() {

        const { open, it_is_an_error_message, title, message, onclick, fullScreen, classes } = this.props;

        // Only if it's an error message we display the ErrorOutline icon

        return (

            <Dialog
                open={open}
                onClose={this.handleClose}
                fullScreen={fullScreen}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
                className={classes.dialog}>
                { it_is_an_error_message ?
                    (
                        <DialogTitle className={classes.dialog_title}>
                            <ErrorOutline className={classes.error_outline} />
                        </DialogTitle>
                    ) : ( <React.Fragment></React.Fragment> )
                }
                <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
                <DialogContent>
                    <DialogContentText id="alert-dialog-description">
                        {message}
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button onClick={onclick} color="primary" autoFocus>
                        OK
                    </Button>
                </DialogActions>
            </Dialog>

        )

    }

}

export default withMobileDialog()(withStyles(styles)(MessageDialog));
