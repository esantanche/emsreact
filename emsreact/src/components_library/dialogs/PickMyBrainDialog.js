import React, { Component } from 'react';

// Material UI components
import Button from "@material-ui/core/es/Button/Button";

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import withMobileDialog from '@material-ui/core/withMobileDialog';

import PropTypes from 'prop-types'

import ErrorOutline from '@material-ui/icons/ErrorOutline';
import AppBar from "@material-ui/core/AppBar/AppBar";
import Toolbar from "@material-ui/core/Toolbar/Toolbar";
import IconButton from "@material-ui/core/IconButton/IconButton";
import PickMyBrainScreen from "../../pick_my_brain/PickMyBrainScreen";
import CardContent from "@material-ui/core/CardContent";
import {withStyles} from "@material-ui/core/styles";

import CloseIcon from '@material-ui/icons/Close';
import CardHeader from "@material-ui/core/es/CardHeader/CardHeader";
import ColouredCard from "../cards/ColouredCard";


// ./src/components_library/dialogs/PickMyBrainDialog.js
// Line 73:  'classes' is not defined    no-undef
// Line 76:  'CloseIcon' is not defined  react/jsx-no-undef
//
// Search for the keywords to learn more about each error.

// Is position relative actually needed?
const styles = {
    appBar: {
        position: 'relative',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'flex-end'
    }
};

class PickMyBrainDialog extends Component {

    // FIXME needed?
    // static propTypes = {
    //     open: PropTypes.bool.isRequired,
    //     it_is_an_error_message: PropTypes.bool,
    //     title: PropTypes.string,
    //     message: PropTypes.string,
    //     onclick: PropTypes.func,
    //     fullScreen: PropTypes.bool.isRequired
    // };

    static propTypes = {
        open: PropTypes.bool.isRequired,
        onclick: PropTypes.func
    };

    constructor(props) {
        super(props);

        this.state = {
            // The element the menu is anchored to, typically a button that opens the menu
            // When anchorElement is null, the menu is invisible
            // FIXME ? ok?
            open: false
        };

    }

    // handleClose = () => {
    //     this.setState({ open: false });
    // };

    render() {

        // FIXME fix all these props

        const { open, it_is_an_error_message, title, message, onclick, fullScreen, classes } = this.props;

        // style={{ display: "flex", flexDirection: "column", alignItems: "center" }}
        // style={{ display: "flex", flexDirection: "row" }}

        return (

            <Dialog
                open={open}
                onClose={this.handleClose}
                fullScreen={fullScreen}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
                >
                <AppBar className={classes.appBar}>
                    <Toolbar>
                        <IconButton color="inherit" onClick={onclick} aria-label="Close">
                            <CloseIcon />
                        </IconButton>
                    </Toolbar>
                </AppBar>

                <ColouredCard light={true}>

                    <CardHeader title="Pick My Brain!"
                                subheader="Or just ask me a question" />

                    <CardContent>

                        <PickMyBrainScreen onclick={onclick} />

                    </CardContent>

                </ColouredCard>

            </Dialog>

        )

    }

}

export default withMobileDialog()(withStyles(styles)(PickMyBrainDialog));


// export default withMobileDialog()(PickMyBrainDialog);



