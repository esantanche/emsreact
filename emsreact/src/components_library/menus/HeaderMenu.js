import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Menu from "@material-ui/core/Menu/Menu";
import MenuItem from "@material-ui/core/MenuItem/MenuItem";

/**
 * Header menu.
 *
 * Props
 * @param {object} anchorElement The element this menu should anchor to.
 * It gives this menu a position.
 * @param {function} handleMenuClose Function to call when the user clicks
 * on a menu item
 */
class HeaderMenu extends Component {

    static propTypes = {
        anchorElement: PropTypes.object,
        handleMenuClose: PropTypes.func
    };

    render() {

        const { anchorElement, handleMenuClose } = this.props;

        // To close this menu, just make anchorElement null

        // Pick My Brain menu item
        // It calls handleMenuClose with no url and true as second parameter
        // This means that the parent component will open the Pick My Brain Dialog
        // See HeaderScreen.js and PickMyBrainDialog.js

        return (

            <Menu
                id="header-menu"
                anchorEl={anchorElement}
                open={Boolean(anchorElement)}
                onClose={handleMenuClose}>

                <MenuItem onClick={() => handleMenuClose("/")}>Home</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/how-i-work")}>How I work</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/the-tools-i-use")}>The tools I use</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/success-stories")}>Success stories</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/testimonials")}>Testimonials</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/talking-about-my-experiences")}>Articles</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/where-how")}>Where? How?</MenuItem>
                <MenuItem onClick={() => handleMenuClose("", true)}>
                    Pick My Brain!
                </MenuItem>

            </Menu>

        )

    }

}

export default HeaderMenu;


