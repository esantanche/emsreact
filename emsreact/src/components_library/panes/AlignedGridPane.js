import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Grid from '@material-ui/core/Grid';

// Material UI function that make classes available
import { withStyles } from "@material-ui/core/styles/index";

const styles = {
    AlignedGridPane: {
        display: "flex",
        flexDirection: "row",
        alignItems: "center"
    },
};

/**
 * This is a pane. So it's used to position other components.
 * This one is a grid that aligns its children vertically.
 *
 * Props
 * @param {object} classes Classes injected by withStyles function
 */
class AlignedGridPane extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired,
    };

    render() {

        const { classes } = this.props;

        return (

            <Grid container spacing={0} className={classes.AlignedGridPane}>

                {this.props.children}

            </Grid>

        )

    }

}

export default withStyles(styles)(AlignedGridPane);
