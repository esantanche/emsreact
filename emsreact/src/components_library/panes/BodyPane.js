import React, { Component } from 'react';
import { withStyles } from "@material-ui/core/styles/index";
import PropTypes from 'prop-types';

const styles = {
    bodyPane: {
        backgroundColor: "#FFFFFF",
        overflowY: "hidden"
    },
};

class BodyPane extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired
    };

    render() {

        const { classes } = this.props;

        return (

            <div className={classes.bodyPane}>

                {this.props.children}

            </div>

        )

    }

}

export default withStyles(styles)(BodyPane);
