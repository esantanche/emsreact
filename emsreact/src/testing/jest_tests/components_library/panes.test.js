// FIXME docs

// import ReactDOM from "react-dom";
import React from "react";
import renderer from 'react-test-renderer';
//import MasterAvatar from "../../../components_library/avatars/MasterAvatar";

import red from "@material-ui/core/colors/red";
import MasterAppBar from "../../../components_library/bars/MasterAppBar";
import MenuButton from "../../../components_library/buttons/MenuButton";

import { shallow } from 'enzyme';
import MessageDialog from "../../../components_library/dialogs/MessageDialog";
import ArticleCard from "../../../components_library/cards/ArticleCard";

import { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import PickMyBrainDialog from "../../../components_library/dialogs/PickMyBrainDialog";
import HeaderMenu from "../../../components_library/menus/HeaderMenu";
import ReactDOM from "react-dom";
import AlignedGridPane from "../../../components_library/panes/AlignedGridPane";
import BodyPane from "../../../components_library/panes/BodyPane";
import Typography from "@material-ui/core/Typography";
import ButtonBase from "@material-ui/core/ButtonBase/ButtonBase";
import BottomGridPane from "../../../components_library/panes/BottomGridPane";
import FrontGridPane from "../../../components_library/panes/FrontGridPane";
import HomeButtonPane from "../../../components_library/panes/HomeButtonPane";
import InnerGridPane from "../../../components_library/panes/InnerGridPane";
import Card from "@material-ui/core/Card";
import SeparatorPane from "../../../components_library/panes/SeparatorPane";

// FIXME evertyhing

// FIXME Here we test components in components_library/avatars

{/*<MessageDialog open={this.state.error_dialog_is_open}*/}
               {/*it_is_an_error_message={true}*/}
               {/*title="Something went wrong"*/}
               {/*message="Please, save your message, reload the page and try again."*/}
               {/*onclick={this.handleOnClickErrorDialog.bind(this)}/>*/}

{/*<MessageDialog open={this.state.message_dialog_is_open}*/}
{/*it_is_an_error_message={false}*/}
{/*title="Message sent"*/}
{/*message="Thank you for your message!"*/}
{/*onclick={this.handleOnClickDialog.bind(this)}/>*/}

{/*<MessageDialog open={this.state.message_dialog_is_open}*/}
               {/*it_is_an_error_message={false}*/}
               {/*title="Message sent"*/}
               {/*message="Thank you for your message!"*/}
               {/*onclick={this.handleOnClickDialog.bind(this)}/>*/}


//MessageDialog

// FIXME

describe('Testing AlignedGridPane', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
                {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    // const mockHandle = jest.fn();
    //
    // const div = document.createElement('div');

    const aligned_grid_pane_to_be_tested = ( <AlignedGridPane /> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('AlignedGridPane renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(aligned_grid_pane_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});

describe('Testing BodyPane', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
    {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    // const mockHandle = jest.fn();
    //
    // const div = document.createElement('div');

    const body_pane_to_be_tested = ( <BodyPane /> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('BodyPane renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(body_pane_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});

describe('Testing BottomGridPane', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
    {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    // const mockHandle = jest.fn();
    //
    // const div = document.createElement('div');

    const bottom_grid_pane_to_be_tested = ( <BottomGridPane
        leftComponent={ (

            <React.Fragment>

                <Typography variant="title" >

                    Emanuele Santanch√©

                </Typography>

                <Typography variant="subheading" >

                    Paris, France and, remotely, worldwide

                </Typography>

            </React.Fragment>

        ) }
        rightComponent={ (

            <ButtonBase href="/article/58/privacy-policy">
                <Typography variant="title" >

                    Privacy policy

                </Typography>
            </ButtonBase>

        ) }
    /> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('BottomGridPane renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(bottom_grid_pane_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});


describe('Testing FrontGridPane', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
    {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    // const mockHandle = jest.fn();
    //
    // const div = document.createElement('div');

    const front_grid_pane_to_be_tested = ( <FrontGridPane /> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('FrontGridPane renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(front_grid_pane_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});


describe('Testing HomeButtonPane', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
    {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    // const mockHandle = jest.fn();
    //
    // const div = document.createElement('div');

    const home_button_pane_to_be_tested = ( <HomeButtonPane /> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('HomeButtonPane renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(home_button_pane_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});


describe('Testing InnerGridPane', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
    {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    // const mockHandle = jest.fn();
    //
    // const div = document.createElement('div');

    const inner_grid_pane_to_be_tested = ( <InnerGridPane key={1}

                                                          leftComponent={ (

                                                              <div>left component</div>

                                                          ) }

                                                          rightComponent={ (

                                                              <div>right component</div>

                                                          ) }

    /> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('InnerGridPane renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(inner_grid_pane_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});


describe('Testing SeparatorPane', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
    {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    // const mockHandle = jest.fn();
    //
    // const div = document.createElement('div');

    const separator_pane_to_be_tested = ( <SeparatorPane /> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('SeparatorPane renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(separator_pane_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});