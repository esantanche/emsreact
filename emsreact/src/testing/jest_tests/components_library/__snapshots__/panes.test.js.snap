// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AlignedGridPane AlignedGridPane renders correctly 1`] = `
<div
  className="MuiGrid-container-2 AlignedGridPane-AlignedGridPane-1"
/>
`;

exports[`Testing BodyPane BodyPane renders correctly 1`] = `
<div
  className="BodyPane-bodyPane-98"
/>
`;

exports[`Testing BottomGridPane BottomGridPane renders correctly 1`] = `
<div
  className="MuiGrid-container-2 MuiGrid-spacing-xs-8-23 BottomGridPane-container-99"
>
  <div
    className="MuiGrid-item-3 MuiGrid-grid-xs-9-38"
  >
    <h2
      className="MuiTypography-root-101 MuiTypography-title-107"
    >
      Emanuele Santanché
    </h2>
    <h3
      className="MuiTypography-root-101 MuiTypography-subheading-108"
    >
      Paris, France and, remotely, worldwide
    </h3>
  </div>
  <div
    className="MuiGrid-item-3 MuiGrid-grid-xs-3-32 BottomGridPane-right_element-100"
  >
    <a
      className="MuiButtonBase-root-125"
      href="/article/58/privacy-policy"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <h2
        className="MuiTypography-root-101 MuiTypography-title-107"
      >
        Privacy policy
      </h2>
      <span
        className="MuiTouchRipple-root-128"
      />
    </a>
  </div>
</div>
`;

exports[`Testing FrontGridPane FrontGridPane renders correctly 1`] = `
<div
  className="MuiGrid-container-2 MuiGrid-spacing-xs-16-24"
>
  <div
    className="MuiGrid-item-3 MuiGrid-grid-md-1-58 MuiGrid-grid-lg-2-73"
  />
  <div
    className="MuiGrid-item-3 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-10-67 MuiGrid-grid-lg-8-79"
  />
  <div
    className="MuiGrid-item-3 MuiGrid-grid-md-1-58 MuiGrid-grid-lg-2-73"
  />
</div>
`;

exports[`Testing HomeButtonPane HomeButtonPane renders correctly 1`] = `
<div
  className="HomeButtonPane-HomeButtonPane-135 HomeButtonPane-TopAndBottomMargins-136"
/>
`;

exports[`Testing InnerGridPane InnerGridPane renders correctly 1`] = `
<div
  className="InnerGridPane-InnerGridPane-137"
>
  <div
    className="MuiGrid-container-2 MuiGrid-spacing-xs-16-24"
  >
    <div
      className="MuiGrid-item-3 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-12-55 MuiGrid-grid-md-6-63 MuiGrid-grid-lg-6-77"
    >
      <div>
        left component
      </div>
    </div>
    <div
      className="MuiGrid-item-3 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-12-55 MuiGrid-grid-md-6-63 MuiGrid-grid-lg-6-77"
    >
      <div>
        right component
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing SeparatorPane SeparatorPane renders correctly 1`] = `
<div
  className="SeparatorPane-separatorPane-138"
>
   
</div>
`;
